### ISSUE TITLE
Client Side exception Error in v2

### ISSUE THREAD

**Issue by dhwaj1902** [umami-software/umami#1943]
Suddenly, umami is showing ``Application error: a client-side exception has occurred (see the browser console for more information).``

Using self-hosted umami. Below is the image for the reference.

![image](https://user-images.githubusercontent.com/57860123/233499399-26b65106-db92-49ec-9794-a0298190116e.png)

**Comment by mikecao** [comment]
Need to know which page.

**Comment by dhwaj1902** [comment]
It was Main Page.

**Comment by pierre-vassoilles** [comment]
Hi! 
I had the same problem today (on all pages) after migrating from 1.40 to 2.0. I use node 20.0.0 and yarn 1.22.19.
I managed to fix it by removing all cookies and local storage (from dev tools), so I think the bug comes from stored data.

I'll let you know if the problem returns ;)

**Comment by Reduction77** [comment]
the same, I have I have reinstalled umami.

**Comment by Reduction77** [comment]
He's getting better again without even realizing it. It seems to be in fits and starts😂

**Comment by dhwaj1902** [comment]
> Hi! I had the same problem today (on all pages) after migrating from 1.40 to 2.0. I use node 20.0.0 and yarn 1.22.19. I managed to fix it by removing all cookies and local storage (from dev tools), so I think the bug comes from stored data.
> 
> I'll let you know if the problem returns ;)

Yes it will definitely resolve the issue, but when you reload, the issue persists.

**Comment by vFire** [comment]
A lot of bugs seemed, after the server side is correctly running, once I visit the website be embed the tracking code, the server just borken.

umami/node_modules/mmdb-lib/lib/decoder.js:52 throw new Error('Invalid Extended Type at offset ' + offset + ' val ' + tmp);

**Comment by maielo** [comment]
+1

**Comment by runtop** [comment]
+1

**Comment by mikecao** [comment]
A lot of these issues are related to your Node version. Try updating to the latest version and see if it still errors.

**Comment by runtop** [comment]
> A lot of these issues are related to your Node version. Try updating to the latest version and see if it still errors.

server's env : ubuntu server, with nodejs 18.16.0 and umami version is 2.1.1
everything is OK after open admin site ，and navigate fine by button on pages.
but just only manual refresh full web page by F5  (any page) ,the browser will show js error.

**Comment by anseltaft** [comment]
+1 on the latest version of Umami - but it worked as expected in incognito mode so I'll try to clear the local storage

**Comment by anseltaft** [comment]
> A lot of these issues are related to your Node version. Try updating to the latest version and see if it still errors.

So far this seems to have fixed the problem. For anyone else reading this here's what I did:

`docker compose down `

to bring down the docker compose stack.

Then I nuked the umami image in Portainer (which can also be done in the command line)

Then I edited my docker compose yaml file to add a new environment declaration:

`nano docker-compose.yml`

Docker-Compose.yml:

```
services:
  umami:
    image: docker.umami.dev/umami-software/umami:postgresql-latest
    container_name: umami
    ports:
      - 127.0.0.1:3000:3000
    environment:
      DATABASE_URL: postgresql://umami:umami@db:5432/umami
      DATABASE_TYPE: postgresql
      HASH_SALT: <add your salt string here>
      TRACKER_SCRIPT_NAME: custom-tracker-name-here.js
      IGNORE_IP: <insert-your-home-and-or-work-ips-here-separated-by-commas>
      NODE_VERSION: 20.0.0
    labels:
      - traefik.enable=true
      - traefik.http.routers.umami.rule= 
      - traefik.http.routers.umami.entrypoints=websecure
      - traefik.http.routers.umami.tls.certresolver=myresolver
    depends_on:
      - db
    restart: unless-stopped
    logging:
      options:
        max-size: "10m"
        max-file: "10"
  db:
    image: postgres:12-alpine
    container_name: umami-db
    environment:
      POSTGRES_DB: umami
      POSTGRES_USER: umami
      POSTGRES_PASSWORD: umami
    volumes:
      - ./sql/schema.postgresql.sql:/docker-entrypoint-initdb.d/schema.postgresql.sql:ro
      - umami-db-data:/var/lib/postgresql/data
    restart: unless-stopped
    logging:
      options:
        max-size: "10m"
        max-file: "10"
volumes:
  umami-db-data:
```

Be sure to remove the Traefik section if you don't want/use Traefik as a front-end proxy. I also nuked everything the `traefik.http.routers.umami.rule=` declaration because that's where you insert your sub/domain.

Cheers

**Comment by runtop** [comment]
thanks for your reply!




but i installed umami throught pull down all source code, then how to fixed this ?


i used umami version : 2.2.0
nodejs version: v18.16.0
















At 2023-05-02 08:21:00, "Ansel Taft" ***@***.***> wrote:

A lot of these issues are related to your Node version. Try updating to the latest version and see if it still errors.

So far this seems to have fixed the problem. For anyone else reading this here's what I did:

docker compose down

to bring down the docker compose stack.

Then I nuked the umami image in Portainer (which can also be done in the command line)

Then I edited my docker compose yaml file to add a new environment declaration:

nano docker-compose.yml

Docker-Compose.yml:

services:
  umami:
    image: docker.umami.dev/umami-software/umami:postgresql-latest
    container_name: umami
    ports:
      - 127.0.0.1:3000:3000
    environment:
      DATABASE_URL: ***@***.***:5432/umami
      DATABASE_TYPE: postgresql
      HASH_SALT: <add your salt string here>
      TRACKER_SCRIPT_NAME: custom-tracker-name-here.js
      IGNORE_IP: <insert-your-home-and-or-work-ips-here-separated-by-commas>
      NODE_VERSION: 20.0.0
    labels:
      - traefik.enable=true
      - traefik.http.routers.umami.rule= 
      - traefik.http.routers.umami.entrypoints=websecure
      - traefik.http.routers.umami.tls.certresolver=myresolver
    depends_on:
      - db
    restart: unless-stopped
    logging:
      options:
        max-size: "10m"
        max-file: "10"
  db:
    image: postgres:12-alpine
    container_name: umami-db
    environment:
      POSTGRES_DB: umami
      POSTGRES_USER: umami
      POSTGRES_PASSWORD: umami
    volumes:
      - ./sql/schema.postgresql.sql:/docker-entrypoint-initdb.d/schema.postgresql.sql:ro
      - umami-db-data:/var/lib/postgresql/data
    restart: unless-stopped
    logging:
      options:
        max-size: "10m"
        max-file: "10"
volumes:
  umami-db-data:


Be sure to remove the Traefik section if you don't want/use Traefik as a front-end proxy. I also nuked everything the traefik.http.routers.umami.rule= declaration because that's where you insert your sub/domain.

Cheers

**Comment by anseltaft** [comment]
> thanks for your reply! but i installed umami throught pull down all source code, then how to fixed this ? i used umami version : 2.2.0 nodejs version: v18.16.0

If you installed from source you'll want to update NodeJS from the commandline:
https://blog.hubspot.com/website/update-node-js
